src/Color/Scale.purs

```diff
@@ -53,19 +64,26 @@ stopColor :: ColorStop -> Color
 stopColor (ColorStop c _) = c
 
 -- | A color scale.
-data ColorScale = ColorScale ColorSpace Color (List ColorStop) Color
+data ColorScaleBuilder = ColorScaleBuilder Color (List ColorStop) Color
+data ColorScale = ColorScale ColorSpace ColorScaleBuilder
 
 -- | Create a color scale. The color space is used for interpolation between
 -- | different stops. The first `Color` defines the left end (color at ratio
 -- | 0.0), the list of stops defines possible intermediate steps and the second
 -- | `Color` argument defines the right end point (color at ratio 1.0).
 colorScale :: ColorSpace -> Color -> List ColorStop -> Color -> ColorScale
-colorScale = ColorScale
+colorScale space b middle e = ColorScale space $ colorScale' b middle e
+
+colorScale' :: Color -> List ColorStop -> Color -> ColorScaleBuilder
+colorScale' = ColorScaleBuilder
 
 -- | Create a uniform color scale from a list of colors that will be evenly
 -- | spaced on the scale.
 uniformScale :: forall f. Foldable f => ColorSpace -> Color -> f Color -> Color -> ColorScale
-uniformScale mode b middle e = colorScale mode b stops e
+uniformScale mode b middle e = ColorScale mode $ uniformScale' b middle e
+
+uniformScale' :: forall f. Foldable f => Color -> f Color -> Color -> ColorScaleBuilder
```

Comment:
this `'` functions might be useful when someone is working on ColorScaleBuilder and might not care about ColorSpace. Like I did [here](https://github.com/safareli/purescript-colorpalettepicker-halogen/blob/9b47d80d1bec6a5e44eb1361045b24166d0d0e35/src/ColorPalettePicker/Palettes.purs#L294)

Reply:
Okay. Can we please add a docstring?